doc/automated-init.md

```diff
@@ -0,0 +1,113 @@
+
+To skip this interactive step we need to buid this configuration once and save it to the JSON file for later reuse.
+
+Let's say we will place it into the `CONFIG_FILE` path, like _./resin-os/raspberrypi3-config.json_.
+
+We also need to put the OS image somewhere, let's call this path `OS_IMAGE_PATH`, it cam be something like _./resin-os/raspberrypi3-v2.0.6+rev1.prod.img_.
+
+1. First we need to download the OS image once. That's needed for building the config, and will speedup the consequent operations as the downloaded OS image is placed into the local cache.
+
+	Run:
+	```bash
+	resin os download DEVICE_TYPE --output OS_IMAGE_PATH --version OS_VERSION
+	```
+
+1. Now we're ready to build the config:
+
+	```bash
+	resin os build-config OS_IMAGE_PATH DEVICE_TYPE --output CONFIG_FILE
+	```
+
+	This will run you through the interactive configuration wizard and in the end save the generated config as `CONFIG_FILE`. You can then verify it's not empty:
+
+	```bash
+	cat CONFIG_FILE
+	```
+
+## Done
+
+Now you're ready to run the command in the beginning of this guide.
+
+Please not again that all of these steps only need to be done once (unless you need to change something), and once all the parameters are collected the main init command can be run unchanged.
+
+But there are still some nuances to cover, please read below. 
+
+## Nuances
+
+### `sudo` password on *nix systems
+
+In order to write the image to the raw device we need the root permissions, this is unavoidable.
+
+To improve the security we only run the minimal subcommand with `sudo`.
+
+This means that with the default setup you're interrupted closer to the end of the device init process to enter your sudo password for this subcommand to work.
+
+There are several ways to eliminate it and make the process fully non-interactive.
+
+#### Option 1: make passwordless sudo.
+
+Obviously you shouldn't do that if the machine you're working on has access to any sensitive resources or information.
+
+But if you're using a machine dedicated to resin provisioning this can be fine, and also the simplest thing to do.
+
+#### Option 2: `NOPASSWD` directive
+
+You can configure the `resin` CLI command to be sudo-runnable without the password. Check [this post](https://askubuntu.com/questions/159007/how-do-i-run-specific-sudo-commands-without-a-password) for an example.
+
+### Extra initialization config
+
+As of June 2017 all the supported devices should not require any other interactive configuration.
+
+But by the design of our system it is _possible_ (though it doesn't look very likely it's going to happen any time soon) that some extra initialization options may be requested for the specific device types.
+
+If that is the case please raise the issue in the resin CLI repository and the maintainers will add the necessary options to build the similar JSON config for this step.
```

Comment:
It's hard to handle because we often have lots of subcommands called from the original command.
There's no universal `--non-interactive` switch. `--config` covers the configuration step (see explanation above), and `--yes` only covers the _confirmations_ (Are you sure you want to explode your machine?). But if the info is missing right now the nested commands have no indication they should bail because they're operating in non-interactive mode.